istioctl/cmd/analyze.go

```diff
@@ -123,6 +130,8 @@ istioctl experimental analyze -k a.yaml b.yaml
 
 	analysisCmd.PersistentFlags().BoolVarP(&useKube, "use-kube", "k", false,
 		"Use live kubernetes cluster for analysis")
+	analysisCmd.PersistentFlags().StringVarP(&useDiscovery, "discovery", "d", "",
+		"'true' or 'false' explicitly enables/disables service discovery. Otherwise, default to enabled if use-kube is set")
```

Comment:
Simplify the usage.  Perhaps `Enable service discovery (default true if --use-kube)`?

Reply:
Took a stab at this. Let me know if you think it still needs tweaking.